core/cap-0051.md

```diff
@@ -0,0 +1,145 @@
+```
+CAP: 0050
+Title: Smart Contract Host Functionality: Secp256r1 Verification
+Working Group:
+    Owner: Leigh McCulloch <@leighmcculloch>
+    Authors: Leigh McCulloch <@leighmcculloch>
+    Consulted: Graydon Hoare <@graydon>, Riad S. Wahby <@kwantam>
+Status: Draft
+Created: 2023-01-30
+Discussion: https://discord.com/channels/897514728459468821/1199160204676300841
+Protocol version: TBD
+```
+
+## Simple Summary
+
+Support secp256r1 verification in Soroban contracts via the exported host
+interface.
+
+## Motivation
+
+Secp256r1, also commonly known as prime256r1 or ecdsa p256, is a common elliptic
+curve used with ecdsa. It's one of the common signature algorithms used in
+webauthn, which is the standard behind passkeys available on, browsers,
+computers, and phones. Supporting secp256r1 natively would allow for the
+execution of efficient webauthn implementations where-by browsers, computers,
+and phones could be the signers of accounts on Stellar via Soroban's custom
+account interface. While it is possible to embed ecdsa secp256r1 verification
+into contracts on the guest side, the instruction cost of ecdsa verification is
+greater than the networks current maximum limits.
+
+### Goals Alignment
+
+This CAP is aligned with the following Stellar Network Goals:
+
+- The Stellar Network should make it easy for developers of Stellar projects to
+  create highly usable products
+
+## Abstract
+
+One function `verify_sig_ecdsa_secp256r1` is added to the Soroban environment's
+exported function interface that accepts the parameters needed to verify a
+secp256r1 signature of a payload for a public key.
+
+## Specification
+
+### XDR Changes
+
+None.
+
+### Host Function Changes
+
+```diff mddiffcheck.ignore=true mddiffcheck.base=v20.1.0
```

Comment:
I'll have to come back and make it possible for this check to work, but right now it won't which is why it is set to ignore.